### DISCUSSION TITLE
Semantics of manifest properties: "package" == manifest vs. "package" == software

### DISCUSSION THREAD

**Discussion by marco-eckstein** [microsoft/winget-cli#4014] [Q&A] [answered]
### Brief description of your issue

I think the descriptions of some manifest properties should be more clear regarding the semantics of the word _package_ - whether it refers to the manifest or the actual software.

There are some properties that are clearly manifest properties, e.g.:

- PackageIdentifier (a concept that only exists for WinGet)
  "description": "The **package** unique identifier"
- PackageLocale
  "description": "The **package meta-data** locale"

Then there are some that are clearly software properties, e.g.:

- PackageVersion
  "description": "The **package** version"
- Publisher
  "description": "The publisher name"

You can see that the descriptions are not fully consistent. And what about the following properties?

- Author
   "description": "The package author"
- PackageUrl
  "description": "The package home page"

Since it is clear that _Publisher_ is the publisher of the software and not (necessarily) the manifest, I thought _Author_ would refer to the author of the manifest. But now I have come to think that _Author_ refers to the author of the software. However, what is the difference then between _Author_ and _Publisher_? Can you give an example where they would be different?

And does _PackageUrl_ refer to the software's home page?

**Comment by denelon** [comment, marked as answer]
"Author" does refer to the software/package, and so does "Publisher". I don't remember off the top of my head any packages where these differ, but if memory serves correctly, there were a couple of cases where the author is an individual, but an organization actually publishes the package.

The manifest schema has evolved over time, and rather than making breaking changes, we've decided to keep the original field names, and to clarify, when possible, as new fields are added. If/when we get to a breaking change release, we would likely revisit all the manifest fields for the sake of consistency.
